cmd/gather/README.md

```diff
@@ -0,0 +1,36 @@
+# OpenTelemetry Operator Must-Gather
+
+The OpenTelemetry Operator `must-gather` tool is designed to collect comprehensive information about OpenTelemetry components within an OpenShift cluster. This utility extends the functionality of [OpenShift must-gather](https://github.com/openshift/must-gather) by specifically targeting and retrieving data related to the OpenTelemetry Operator, helping in diagnostics and troubleshooting.
+
+Note that you can use this utility too to gather information about the objects deployed by the OpenTelemetry Operator if you don't use OpenShift.
+
+## What is a Must-Gather?
+
+The `must-gather` tool is a utility that collects logs, cluster information, and resource configurations related to a specific operator or application in an OpenShift cluster. It helps cluster administrators and developers diagnose issues by providing a snapshot of the cluster's state related to the targeted component. More information [in the official documentation](https://docs.openshift.com/container-platform/4.16/support/gathering-cluster-data.html).
+
+## Usage
+
+First, you will need to build and push the image:
+```sh
+make container-must-gather container-must-gather-push
```

Comment:
For the record, I do think we should publish this image (and probably version it the same way we do the operator?) and not force users to build it themselves.

Reply:
Ok, I can do. Just wanted to avoid maintaining another image.

Reply:
Well, how much maintenence the image actually needs depends on how elaborate we make the release process. I think it's simplest to use the same versioning as we do for the operator, but I can also see building it on each commit if the files have changed, and just tagging it with the commit hash. Realistically, users will always want the latest one after all.

Reply:
We can do it per release. I just applied the changes. Thanks for the feedback!